composer/datasets/in_context_learning_evaluation.py

```diff
@@ -300,11 +300,11 @@ def prep_examples(self, num_fewshot: int, prompt_string: str, example_delimiter:
                 query = f'{example_delimiter}{query}'
 
             choices = [f'{continuation_delimiter}{choice}' for choice in choices]
-            encoded_example['query'] = self.tokenizer(query)
-            encoded_example['choices'] = [self.tokenizer(choice) for choice in choices]
             encoded_example['preamble'] = self.tokenizer(
                 preamble)  # if the preamble is empty then these will be 0-length lists
             encoded_example['gold_idx'] = gold_idx
+            encoded_example['query'] = self.tokenizer(query, add_special_tokens=len(preamble) == 0)
```

Comment:
same comment as above.

Reply:
Good catch, I was actually prepending two <s/> tokens in the case of zero-shot eval, so it is fixed now!

Reply:
why was it ending up with doubled special tokens? just for my undestanding. it looks like you would have one at the start of preamble (whether preamble was empty or not), and then where would the other one come from?